celery_prometheus_exporter.py

```diff
@@ -48,26 +49,40 @@ def _process_event(self, evt):
         # with this exact situation so we'll use that for now.
         with self._state._mutex:
             if evt['type'].startswith('task-'):
-                self._collect_tasks(evt)
+                event_state = evt['type'].split('-').pop()
+                state = celery.events.state.TASK_EVENT_TO_STATE[event_state]
+                if state == celery.states.STARTED:
+                    self._observe_latency(evt)
+                self._collect_tasks(evt, state)
             WORKERS.set(len([w for w in self._state.workers.values() if w.alive]))
 
-    def _process_task_received(self, evt):
-        with self._state._mutex:
-            self._tasks_started[evt['uuid']] = evt['local_received']
-            self._collect_tasks(evt)
-
-    def _process_task_started(self, evt):
-        with self._state._mutex:
-            try:
-                start = self._tasks_started.pop(evt['uuid'])
-            except KeyError:
-                pass
-            else:
-                LATENCY.observe(evt['local_received'] - start)
-            self._collect_tasks(evt)
-
-    def _collect_tasks(self, evt):
-        self._state._event(evt)
+    def _observe_latency(self, evt):
+        try:
+            prev_evt = self._state.tasks[evt['uuid']]
+        except KeyError:
+            pass
+        else:
+            # ignore latency if it is a retry
+            if prev_evt.state == celery.states.RECEIVED:
```

Comment:
If the KeyError actually happens, wouldn't that check result in an NameError due to `prev_evt` not being defined?